src/modelframe.jl

```diff
@@ -131,31 +131,67 @@ function evalcontrasts(df::AbstractDataFrame, contrasts::Dict = Dict())
     return evaledContrasts
 end
 
+
+## copied from DataFrames:
+function _nonmissing!(res, col)
+    # workaround until JuliaLang/julia#21256 is fixed
+    eltype(col) >: Missing || return
+    
+    @inbounds for (i, el) in enumerate(col)
+        res[i] &= !ismissing(el)
+    end
+end
+
+function _nonmissing!(res, col::CategoricalArray{>: Missing})
+    for (i, el) in enumerate(col.refs)
+        res[i] &= el > 0
+    end
+end
+
+# handle some inconsistencies between how named tuples are constructed
+if isdefined(Core, :NamedTuple)
+    _select(d::Data.Table, cols::NTuple{N,Symbol} where N) = NamedTuple{cols}(d)
+    _filter(d::T, rows) where T<:Data.Table = T(([col[rows] for col in d]...))
+else
+    _select(d::Data.Table, cols::NTuple{N,Symbol} where N) = d[cols]
+    _filter(d::T, rows) where T<:Data.Table = T([col[rows] for col in d]...)
+end
+
+_size(d::Data.Table) = size(Data.schema(d))
```

Comment:
This could be a bit faster to just copy the definitions in DataStreams, but this is also fine.